boring/src/ssl/callbacks.rs

```diff
@@ -269,6 +270,67 @@ where
     }
 }
 
+unsafe fn to_uninit<'a, T: 'a>(ptr: *mut T) -> &'a mut MaybeUninit<T> {
+    assert!(!ptr.is_null());
+    unsafe { &mut *ptr.cast::<MaybeUninit<T>>() }
+}
```

Comment:
Confirming my understanding: panic on `assert` seems appropriate because otherwise we have UB.